Tests/AppTests/QueryPerformanceTests.swift

```diff
@@ -97,6 +97,14 @@ class QueryPerformanceTests: XCTestCase {
         try await assertQueryPerformance(query, expectedCost: 3900, variation: 150)
     }
 
+    func test_Search_query_productTypeFilter() async throws {
+        let filter = try ProductTypeSearchFilter(expression: .init(operator: .is, value: "plugin"))
+        let query = try Search.query(app.db, ["a"], filters: [filter],
+                                     page: 1, pageSize: Constants.resultsPageSize)
+            .unwrap()
+        try await assertQueryPerformance(query, expectedCost: 0, variation: 150)
```

Comment:
We can't actually run this before deploying to staging, because staging doesn't have the required migration yet and so the perf test fails (as do all the other search perf tests).

Once I'm done with the docc related re-analysis of versions we can deploy from the branch and update the cost before merging.

Reply:
Details for reference:

```
ℹ️ TEST:        test_Search_refresh()
ℹ️ COST:        13682.03
ℹ️ EXPECTED:    9800.0 ± 200.0
ℹ️ ACTUAL TIME: 98.161ms
/__w/SwiftPackageIndex-Server/SwiftPackageIndex-Server/Tests/AppTests/QueryPerformanceTests.swift:148: error: QueryPerformanceTests.test_Search_refresh : failed - Total cost of 13682.03 above threshold of 10000.0 (incl variation)

Query plan:

Nested Loop  (cost=9640.49..13682.03 rows=478 width=319) (actual time=53.248..98.161 rows=6864 loops=1)
  Join Filter: (r.package_id = p.id)
  ->  Hash Right Join  (cost=9640.21..13530.45 rows=478 width=271) (actual time=53.103..84.361 rows=6864 loops=1)
        Hash Cond: (pr.version_id = v.id)
        ->  Seq Scan on products pr  (cost=0.00..3526.97 rows=95597 width=58) (actual time=0.012..12.838 rows=95692 loops=1)
        ->  Hash  (cost=9635.52..9635.52 rows=375 width=245) (actual time=53.048..53.051 rows=4691 loops=1)
              Buckets: 8192 (originally 1024)  Batches: 1 (originally 1)  Memory Usage: 1361kB
              ->  Hash Join  (cost=875.27..9635.52 rows=375 width=245) (actual time=6.537..49.707 rows=4691 loops=1)
                    Hash Cond: ((v.package_id = r.package_id) AND ((v.reference ->> 'branch'::text) = r.default_branch))
                    ->  Seq Scan on versions v  (cost=0.00..8356.73 rows=75073 width=195) (actual time=0.014..32.561 rows=75074 loops=1)
                    ->  Hash  (cost=804.91..804.91 rows=4691 width=208) (actual time=6.452..6.453 rows=4691 loops=1)
                          Buckets: 8192  Batches: 1  Memory Usage: 1187kB
                          ->  Seq Scan on repositories r  (cost=0.00..804.91 rows=4691 width=208) (actual time=0.010..4.016 rows=4691 loops=1)
  ->  Index Scan using packages_pkey on packages p  (cost=0.28..0.30 rows=1 width=80) (actual time=0.002..0.002 rows=1 loops=6864)
        Index Cond: (id = v.package_id)
Planning Time: 3.769 ms
Execution Time: 98.576 ms
```